lms/djangoapps/support/views/program_enrollments.py

```diff
@@ -98,3 +106,153 @@ def _validate_and_link(program_uuid_string, linkage_text):
         ]
         errors = [message for message in link_errors.values()]
         return successes, errors
+
+
+class ProgramEnrollmentsInspectorView(View):
+    """
+    The view to search and display the program enrollments
+    information of a learner.
+    """
+    exclude_from_schema = True
+    CONSOLE_TEMPLATE_PATH = 'support/program_enrollments_inspector.html'
+
+    @method_decorator(require_support_permission)
+    def get(self, request):
+        """
+        Based on the query string parameters passed through the GET request
+        Search the data store for information about ProgramEnrollment and
+        SSO linkage with the user.
+        """
+        errors = []
+        edx_username_or_email = request.GET.get('edx_user', '').strip()
+        org_key = request.GET.get('IdPSelect', '').strip()
+        external_user_key = request.GET.get('external_user_key', '').strip()
+        learner_program_enrollments = {}
+        if edx_username_or_email:
+            learner_program_enrollments, error = self._get_account_info(edx_username_or_email)
+            if error:
+                errors.append(error)
+        elif org_key and external_user_key:
+            learner_program_enrollments = {}
+        elif not external_user_key and org_key:
+            errors.append(
+                'You must provide either the edX username or email, or the '
+                'Learner Account Provider and External Key pair to do search!'
+            )
+
+        return render_to_response(
+            self.CONSOLE_TEMPLATE_PATH,
+            {
+                'successes': [],
+                'errors': errors,
+                'learner_program_enrollments': learner_program_enrollments,
+                'org_keys': self._get_org_keys_with_IdP_provider(),
+            }
+        )
+
+    def _get_org_keys_with_IdP_provider(self):
```

Comment:
```suggestion
    def _get_org_keys_with_idp(self):
```
IdP already stands for Identity Provider. Also, lowercase method names.

Reply:
Feedback taken